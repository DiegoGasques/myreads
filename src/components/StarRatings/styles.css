.star-rating {
  font-size: 11px;
  line-height: 1.9;
  color: #999;
}
