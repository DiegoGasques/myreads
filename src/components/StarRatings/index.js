import React from "react";
import PropTypes from "prop-types";
import Ratings from "react-ratings-declarative";

import "./styles.css";

function StarRatings({ avgRating }) {
  return avgRating ? (
    <div className="star-rating">
      <Ratings
        rating={avgRating}
        widgetDimensions="13px"
        widgetSpacings="0.1px"
        widgetRatedColors="#f8ce0b"
      >
        <Ratings.Widget />
        <Ratings.Widget />
        <Ratings.Widget />
        <Ratings.Widget />
        <Ratings.Widget />
      </Ratings>
      <span style={{ fontWeight: "bold" }}> {`${avgRating}`}</span> avg. rating
    </div>
  ) : null;
}

Ratings.propTypes = {
  avgRating: PropTypes.number
};

export default StarRatings;
