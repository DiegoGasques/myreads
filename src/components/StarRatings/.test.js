import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import StarRatings from "./index";

it("Should match StarRatings component snapshot with avgRating prop", () => {
  const wrapper = shallow(<StarRatings avgRating={3.8} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("Should match StarRatings component snapshot without avgRating prop", () => {
  const wrapper = shallow(<StarRatings />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
