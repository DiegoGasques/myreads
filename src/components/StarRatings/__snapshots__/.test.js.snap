// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match StarRatings component snapshot with avgRating prop 1`] = `
<div
  className="star-rating"
>
  <Ratings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    rating={3.8}
    svgIconPaths="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBoxes="0 0 51 48"
    typeOfWidget="Star"
    widgetDimensions="13px"
    widgetEmptyColors="rgb(203, 211, 227)"
    widgetHoverColors="rgb(230, 67, 47)"
    widgetRatedColors="#f8ce0b"
    widgetSpacings="0.1px"
  >
    <Widget />
    <Widget />
    <Widget />
    <Widget />
    <Widget />
  </Ratings>
  <span
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
     
    3.8
  </span>
   avg. rating
</div>
`;

exports[`Should match StarRatings component snapshot without avgRating prop 1`] = `""`;
